modules/Activities/activities_attendanceProcess.php

```diff
@@ -45,6 +45,31 @@
 
     $gibbonPersonID = $_POST['gibbonPersonID'];
 
+    $highestAction = getHighestGroupedAction($guid, '/modules/Activities/activities_attendance.php', $connection2);
+
+    if($highestAction == "Enter Activity Attendance_leader") {
+        try {
+            $dataCheck = array("gibbonPersonID" => $gibbonPersonID, "gibbonActivityID" => $gibbonActivityID);
+            $sqlCheck = "SELECT role FROM gibbonActivityStaff WHERE gibbonActivityID=:gibbonActivityID AND gibbonPersonID=:gibbonPersonID";
+            $resultCheck = $connection2->prepare($sqlCheck);
+            $resultCheck->execute($dataCheck);
+
+            if ($resultCheck->rowCount() > 0) {
+                $row = $resultCheck->fetch();
+                if ($row["role"] != "Organiser" && $row["role"] != "Assistant") {
+                    $URL .= '&return=error0';
+                    header("Location: {$URL}");
```

Comment:
I've added the exits and checked the My Activity Attendance button. The check uses isActionAccessible instead of highestAction, so no problems.